android/app/src/main/java/org/fossasia/openevent/fragments/SponsorsFragment.java

```diff
@@ -147,8 +149,12 @@ public void refreshClicked() {
                 };
                 //show snackbar will be useful if user have blocked notification for this app
                 snackbar = showNotificationSnackBar.showSnackBar();
-                //show notification
-                showNotificationSnackBar.buildNotification();
+                //show notification if device is connected to wifi
+                ConnectivityManager connectivityManager = (ConnectivityManager) getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+                android.net.NetworkInfo isWifi = connectivityManager.getNetworkInfo(ConnectivityManager.TYPE_WIFI);
```

Comment:
Please make a util instead of using the same code again and again in all the files. I believe there is already a util present for this. Please check